server/src/main/java/org/cloudfoundry/identity/uaa/scim/jdbc/JdbcScimGroupProvisioning.java

```diff
@@ -157,6 +157,15 @@ public ScimGroup createOrGet(ScimGroup group, String zoneId) {
         }
     }
 
+    @Override
+    public ScimGroup getOrCreate(ScimGroup group, String zoneId) {
```

Comment:
So we have two implementations: createOrGet and getOrCreate
In the upstream PR https://github.com/cloudfoundry/uaa/pull/668, JdbcScimGroupMembershipManager was changed to use getOrCreate:
https://github.com/GESoftware-CF/uaa/blob/9ec558896229a19d47de455485ec905b1f7b3d8a/server/src/main/java/org/cloudfoundry/identity/uaa/scim/jdbc/JdbcScimGroupMembershipManager.java#L110

Could you confirm which one should be used?

Reply:
checked both upstream and pe_4.30.0 are using createOrGet in JdbcScimGroupMembershipManager.java. both getOrCreate and createOrGet has similar logic, the difference is that createOrGet throws exception of IncorrectResultSizeDataAccessException when there is error; vs. getOrCreate throws EmptyResultDataAccessException or ScimResourceAlreadyExistsException in the error cases.